src/gui/TrackContentObjectView.cpp

```diff
@@ -1313,6 +1419,50 @@ TimePos TrackContentObjectView::draggedTCOPos( QMouseEvent * me )
 }
 
 
+int TrackContentObjectView::knifeMarkerPos( QMouseEvent * me )
+{
+	//Position relative to start of clip
+	const int markerPos = me->pos().x();
+
+	//In unquantized mode, we don't have to mess with the position at all
+	if ( unquantizedModHeld(me) ) { return markerPos; }
+	else
+	{	//Otherwise we...
+		//1: Convert the position to a TimePos
+		const float ppb = m_trackView->trackContainerView()->pixelsPerBar();
+		TimePos midiPos = markerPos * TimePos::ticksPerBar() / ppb;
+		//2: Snap to the correct position, based on modifier keys
+		midiPos = quantizeMarkerPos( midiPos, me->modifiers() & Qt::ShiftModifier );
+		//3: Convert back to a pixel position
+		return midiPos * ppb / TimePos::ticksPerBar();
+	}
+}
+
+
+
+
+TimePos TrackContentObjectView::quantizeMarkerPos( TimePos midiPos, bool shiftMode )
+{
+	const float snapSize = gui->songEditor()->m_editor->getSnapSize();
+	if ( shiftMode )
+	{	//If shift is held we quantize the length of the new left clip...
+		const TimePos leftPos = midiPos.quantize( snapSize );
+		//...or right clip...
+		const TimePos rightOff = m_tco->length() - midiPos;
+		const TimePos rightPos = m_tco->length() - rightOff.quantize( snapSize );
+		//...whichever gives a position closer to the cursor
+		if ( abs(leftPos - midiPos) < abs(rightPos - midiPos) ) { return leftPos; }
+		else { return rightPos; }
+	}
+	else
+	{
+		return (TimePos(midiPos + m_initialTCOPos).quantize( snapSize ) - m_initialTCOPos);
+	}
+}
+
+
+
+
```

Comment:
```suggestion
int TrackContentObjectView::knifeMarkerPos(QMouseEvent * me)
{
	// Position relative to start of clip
	const int markerPos = me->pos().x();

	// In unquantized mode, we don't have to mess with the position at all
	if (unquantizedModHeld(me)) { return markerPos; }
	else
	{	// Otherwise we...
		// 1: Convert the position to a TimePos
		const float ppb = m_trackView->trackContainerView()->pixelsPerBar();
		TimePos midiPos = markerPos * TimePos::ticksPerBar() / ppb;
		// 2: Snap to the correct position, based on modifier keys
		midiPos = quantizeMarkerPos(midiPos, me->modifiers() & Qt::ShiftModifier);
		// 3: Convert back to a pixel position
		return midiPos * ppb / TimePos::ticksPerBar();
	}
}




TimePos TrackContentObjectView::quantizeMarkerPos(TimePos midiPos, bool shiftMode)
{
	const float snapSize = gui->songEditor()->m_editor->getSnapSize();
	if (shiftMode)
	{	// If shift is held we quantize the length of the new left clip...
		const TimePos leftPos = midiPos.quantize(snapSize);
		// ...or right clip...
		const TimePos rightOff = m_tco->length() - midiPos;
		const TimePos rightPos = m_tco->length() - rightOff.quantize(snapSize);
		// ...whichever gives a position closer to the cursor
		if (abs(leftPos - midiPos) < abs(rightPos - midiPos)) { return leftPos; }
		else { return rightPos; }
	}
	else
	{
		return TimePos(midiPos + m_initialTCOPos).quantize( snapSize ) - m_initialTCOPos;
	}
}




```

Parenthesis spacing and `return` statements must not have parenthesis.

Reply:
I'll fix the parenthases around return.

Reply:
Parens fixed in latest commit.